### DISCUSSION TITLE
how to use stream apis to create quic server

### DISCUSSION THREAD

**Discussion by mmedhekar** [aiortc/aioquic#278] [Q&A] [answered]
I am trying to implement a python server using stream apis and was wondering if there any example i can refer? Can someone share how server will create bidirectional stream and start pushing data to client?

**Comment by ahopkins** [comment, marked as answer]
Does this not have what you need?

https://github.com/aiortc/aioquic/blob/main/examples/http3_server.py

**Reply by jlaine** [reply]
I agree with @ahopkins don't go re-inventing a protocol, most likely you can achieve what you want over HTTP/3.

The raw "streams" APIs exposed by the asyncio code are leftovers from the very early days of aioquic when I didn't have a clear sense of what the usecases would be. As noted in #169 I am thinking of removing them from the aioquic API and instead relegating them at best to an example. I really don't like that part of the API, it's too confusing.

**Reply by mmedhekar** [reply]
Thank you very much @ahopkins and @jlaine for the answers. I am trying to implement a push server that can keep pushing data to client after client initiates a connection. was wondering if server can create a stream(s) in response to client request.
